cmd/virt-handler/virt-handler.go

```diff
@@ -229,6 +238,15 @@ func (app *virtHandlerApp) Run() {
 		panic(err)
 	}
 
+	_, err = app.virtCli.ShadowNodeClient().Create(context.TODO(), &v1.ShadowNode{
+		ObjectMeta: metav1.ObjectMeta{
```

Comment:
If you create this object with virt-handler, who will take care of the deletion of this object in case of a node removal?

Reply:
Thanks for pointing this out. We can either use the Kube garbage collector or I can handle this in the shadow node controller.  I don't see any benefit of one over the other, wdyt?

Reply:
By using kube garbage collector, do you mean let the node owns the object?
Shadow-node controller sounds slightly better to me as we can follow the general principle of whoever creates the object will take care of the clean up.

Another question is if the shadow node already exists, do we reconcile its labels to make sure they are up-to-date later?

Reply:
>By using kube garbage collector, do you mean let the node owns the object? 

I think the other way around would be better.

>Another question is if the shadow node already exists, do we reconcile its labels to make sure they are up-to-date later?

I don't follow. Can you elaborate?

Reply:
For example, when you have an upgrade case where some labels are obsoleted, since you already have the old shadow-node there, so you will not recreate it, and then the old shadow-node may have the obsoleted labels.  Are you going to reconcile at some time and remove those obsoleted labels?